internal/indexer/postgres/indexmanifest.go

```diff
@@ -0,0 +1,149 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/rs/zerolog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/pkg/microbatch"
+)
+
+func indexManifest(ctx context.Context, pool *pgxpool.Pool, ir *claircore.IndexReport) error {
+	const (
+		query = `
+		INSERT INTO manifest_index(package_id, dist_id, repo_id, manifest_hash)
+		VALUES ($1, $2, $3, $4)
+		ON CONFLICT DO NOTHING;
+		`
+	)
+	log := zerolog.Ctx(ctx).With().
+		Str("component", "internal/indexer/postgres/indexManifest").
+		Logger()
+
+	if ir.Hash.String() == "" {
+		return fmt.Errorf("received empty hash. cannot associate contents with a manifest hash")
+	}
+	hash := ir.Hash.String()
+
+	records := ir.IndexRecords()
+	if len(records) == 0 {
+		log.Warn().Msg("manifest being indexed has 0 scan records")
+		return nil
+	}
+
+	// obtain a transaction scoped batch
+	tx, err := pool.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("store:indexDistributions failed to create transaction: %v", err)
+	}
+	defer tx.Rollback(ctx)
+
+	queryStmt, err := tx.Prepare(ctx, "queryStmt", query)
+	if err != nil {
+		return fmt.Errorf("failed to create statement: %v", err)
+	}
+
+	mBatcher := microbatch.NewInsert(tx, 500, time.Minute)
+	for _, record := range records {
+		// ignore nil packages
+		if record.Package == nil {
+			continue
+		}
+
+		v, err := toValues(*record)
+		if err != nil {
+			return fmt.Errorf("received a record with an invalid id: %v", err)
+		}
+
+		// if source package exists create record
+		if v[0] != nil {
+			err = mBatcher.Queue(
+				ctx,
+				queryStmt.SQL,
```

Comment:
does this being prepared actually help?

Reply:
I don't think it can *hurt*, but haven't measured. in theory queries like this are good candidates for prepared statements, as we don't want to do the parsing and planning every single time we send the query to PG.